grakn-graql/src/main/java/ai/grakn/graql/internal/parser/QueryVisitor.java

```diff
@@ -726,6 +739,8 @@ private Order getOrder(TerminalNode order) {
                 return ResourceType.DataType.STRING;
             case "boolean":
                 return ResourceType.DataType.BOOLEAN;
+            case "date":
```

Comment:
Just realised, instead of an ugly switch you might be able to iterate over `ResourceType.DataType.SUPPORTED_TYPES` and compare the `dataType.getText()` with the lower case of the keyset. Would also mean you would never need to worry about amending this part of code potentially.